src/styles/shared/components/buttons/_base.scss

```diff
@@ -0,0 +1,118 @@
+@use "sass:map";
+
+@use "@hypothesis/frontend-shared/styles/mixins/focus";
+
+// Set colors for a button
+@mixin _colors($colormap) {
+  color: map.get($colormap, 'foreground');
+  background-color: map.get($colormap, 'background');
+
+  &:disabled {
+    color: map.get($colormap, 'disabled-foreground');
+  }
+}
+
+// Set hover colors and transition for a button
+@mixin _hover-state($colormap) {
+  &:hover:not([disabled]) {
+    color: map.get($colormap, 'hover-foreground');
+    background-color: map.get($colormap, 'hover-background');
+  }
+  transition: color 0.2s ease-out, background-color 0.2s ease-out,
+    opacity 0.2s ease-out;
+}
+
+// Set active state colors for a button
+@mixin _active-state($colormap) {
+  &[aria-expanded='true'],
+  &[aria-pressed='true'] {
+    color: map.get($colormap, 'active-foreground');
+    @if map.get($colormap, 'active-background') {
+      background-color: map.get($colormap, 'active-background');
+    }
+
+    &:hover:not([disabled]) {
+      color: map.get($colormap, 'hover-foreground');
+    }
+
+    &:focus:not([disabled]) {
+      color: map.get($colormap, 'active-foreground');
+    }
+  }
+}
+
+// Variant mixin: may be be used by variants (BEM modifier classes)
+@mixin button--variant($options) {
+  @include _colors(map.get($options, 'colormap'));
+  @if map.get($options, 'withStates') {
+    @include _active-state(map.get($options, 'colormap'));
+    @include _hover-state(map.get($options, 'colormap'));
+  }
+  @content;
+}
+
+// Base mixin for buttons.
+@mixin button($options) {
```

Comment:
This mixin:

* Now supports `--small` and `--large` sizing
* Supports and `inline` option

is otherwise unchanged